connectors/mongo/data_op_processor.go

```diff
@@ -39,7 +40,111 @@ func (mc *BaseMongoConnector) HandleBarrierMessage(barrierMsg iface.DataMessage)
 	return nil
 }
 
-func (mc *BaseMongoConnector) ProcessDataMessage(dataMsg iface.DataMessage) error {
+type dataMsgIdIndex struct {
+	dataMsgId []byte
+	index     int
+}
+
+// returns the new item or existing item, and whether or not a new item was added
+func addToIdIndexMap(m map[int][]*dataMsgIdIndex, dataMsg iface.DataMessage) (*dataMsgIdIndex, bool) {
+	h := hashDataMsgId(dataMsg)
+	items, found := m[h]
+	if found {
+		for _, item := range items {
+			if slices.Equal(item.dataMsgId, *dataMsg.Id) {
+				return item, false
+			}
+		}
+	}
+	item := &dataMsgIdIndex{*dataMsg.Id, -1}
+	m[h] = append(items, item)
+	return item, true
+}
+
+// ProcesDataMessages assumes all have the same Database and Collection
+func (mc *BaseMongoConnector) ProcessDataMessages(dataMsgs []iface.DataMessage) error {
```

Comment:
```suggestion
// ProcessDataMessages assumes all have the same Database and Collection
```

Reply:
Is this an accident?

Reply:
Oh yes. Sorry. Was trying to correct Proces->Process (double 's')